From 1be4a3430ec56442e1cb48c58a74d7eac8b1bafb Mon Sep 17 00:00:00 2001
From: Walter Bender <walter@sugarlabs.org>
Date: Fri, 16 Aug 2013 14:49:53 -0400
Subject: [PATCH 1/2] add age-gender icons

---
 icons/scalable/device/Makefile.am | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/icons/scalable/device/Makefile.am b/icons/scalable/device/Makefile.am
index 7812ff4..25deb4e 100644
--- a/icons/scalable/device/Makefile.am
+++ b/icons/scalable/device/Makefile.am
@@ -43,6 +43,22 @@ icon_DATA =				\
 	drive.svg			\
 	drive-harddisk.svg		\
 	drive-harddisk-usb.svg		\
+	female-0.svg			\
+	female-1.svg			\
+	female-2.svg			\
+	female-3.svg			\
+	female-4.svg			\
+	female-5.svg			\
+	female-6.svg			\
+	female-7.svg			\
+	male-0.svg			\
+	male-1.svg			\
+	male-2.svg			\
+	male-3.svg			\
+	male-4.svg			\
+	male-5.svg			\
+	male-6.svg			\
+	male-7.svg			\
 	media.svg			\
 	media-optical.svg		\
 	media-disk.svg			\
-- 
1.8.1.4

