From f8dff0e071e6bdc32e77700de6fc73f19ac0d7c3 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Fri, 6 Jun 2014 11:12:39 -0400
Subject: [PATCH 2/3] spent time

Signed-off-by: Martin Abente Lahaye <tch@sugarlabs.org>
---
 src/sugar3/activity/activity.py | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/src/sugar3/activity/activity.py b/src/sugar3/activity/activity.py
index 0594b8e..1686c49 100644
--- a/src/sugar3/activity/activity.py
+++ b/src/sugar3/activity/activity.py
@@ -297,6 +297,8 @@ class Activity(Window, Gtk.Container):
         self.connect('delete-event', self.__delete_event_cb)
 
         self._active = False
+        self._active_time = None
+        self._spent_time = 0
         self._activity_id = handle.activity_id
         self.shared_activity = None
         self._join_id = None
@@ -337,6 +339,11 @@ class Activity(Window, Gtk.Container):
                 self._jobject.metadata['launch-times'] = \
                     str(int(time.time()))
 
+            if 'spent-times' in self._jobject.metadata:
+                self._jobject.metadata['spent-times'] += ', 0'
+            else:
+                self._jobject.metadata['spent-times'] = '0'
+
         self.shared_activity = None
         self._join_id = None
 
@@ -390,6 +397,7 @@ class Activity(Window, Gtk.Container):
         jobject.metadata['share-scope'] = SCOPE_PRIVATE
         jobject.metadata['icon-color'] = icon_color
         jobject.metadata['launch-times'] = str(int(time.time()))
+        jobject.metadata['spent-times'] = '0'
         jobject.file_path = ''
 
         # FIXME: We should be able to get an ID synchronously from the DS,
@@ -455,9 +463,17 @@ class Activity(Window, Gtk.Container):
     def get_active(self):
         return self._active
 
+    def _update_spent_time(self, active):
+        if active:
+            self._active_time = time.time()
+        else:
+            self._spent_time += time.time() - self._active_time
+            self._active_time = None
+
     def set_active(self, active):
         if self._active != active:
             self._active = active
+            self._update_spent_time(self._active)
             if not self._active and self._jobject:
                 self.save()
 
@@ -728,6 +744,20 @@ class Activity(Window, Gtk.Container):
             self.metadata['buddies_id'] = json.dumps(buddies_dict.keys())
             self.metadata['buddies'] = json.dumps(self._get_buddies())
 
+        # Accumulate the last spent time.
+        if self._active:
+            self._update_spent_time(False)
+
+        def set_last_value(values_list, new_value):
+            if ', ' not in values_list:
+                return '%d' % new_value
+            else:
+                partial_list = ', '.join(values_list.split(', ')[:-1])
+                return partial_list + ', %d' % new_value
+
+        self.metadata['spent-times'] = set_last_value(
+            self.metadata['spent-times'], self._spent_time)
+
         preview = self.get_preview()
         if preview is not None:
             self.metadata['preview'] = dbus.ByteArray(preview)
-- 
1.9.3

