From f1c5931d7a1449a0b6669b61b0fa408dbb6eeb87 Mon Sep 17 00:00:00 2001
From: Gonzalo Odiard <godiard@gmail.com>
Date: Fri, 14 Feb 2014 09:18:09 -0300
Subject: [PATCH] Hide the password in the cp network proxy section

Signed-off-by: Gonzalo Odiard <godiard@sugarlabs.org>
---
 extensions/cpsection/network/view.py | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/extensions/cpsection/network/view.py b/extensions/cpsection/network/view.py
index d3242dc..36c0bfd 100644
--- a/extensions/cpsection/network/view.py
+++ b/extensions/cpsection/network/view.py
@@ -203,12 +203,19 @@ class StringSettingBox(SettingBox):
     """
     A configuration line for a string setting.
     """
-    def __init__(self, name, setting, setting_key, size_group=None):
+    def __init__(self, name, setting, setting_key, size_group=None,
+                 password_field=False):
         SettingBox.__init__(self, name, size_group)
 
         self._entry = Gtk.Entry()
         self.pack_start(self._entry, True, True, 0)
         self._entry.show()
+        if password_field:
+            self._entry.set_visibility(False)
+            show_password_chk = Gtk.CheckButton(_('Show password'))
+            show_password_chk.connect('toggled', self.__show_pass_toggled_cb)
+            show_password_chk.show()
+            self.pack_start(show_password_chk, False, False, 10)
 
         setting.bind(setting_key, self._entry, 'text',
                      Gio.SettingsBindFlags.NO_SENSITIVITY)
@@ -216,6 +223,9 @@ class StringSettingBox(SettingBox):
     def clear(self):
         self._entry.set_text('')
 
+    def __show_pass_toggled_cb(self, check_button):
+        self._entry.set_visibility(check_button.get_active())
+
 
 class Network(SectionView):
     def __init__(self, model, alerts):
@@ -726,7 +736,7 @@ class Network(SectionView):
         self.box_password = StringSettingBox(
             _('Password:'),
             self._proxy_settings['org.gnome.system.proxy.http'],
-            'authentication-password', size_group)
+            'authentication-password', size_group, password_field=True)
 
         auth_contents_box.pack_start(self.box_password, False, False, 0)
         self.box_password.show()
-- 
1.8.1.4

