From ab8db6f6e861e169feed889573d95d110a4e3cd9 Mon Sep 17 00:00:00 2001
From: Gonzalo Odiard <godiard@gmail.com>
Date: Thu, 12 Dec 2013 00:35:23 -0300
Subject: [PATCH] Connect to hidden network if needed at cp section close

Signed-off-by: Gonzalo Odiard <gonzalo@laptop.org>
---
 extensions/cpsection/network/view.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/extensions/cpsection/network/view.py b/extensions/cpsection/network/view.py
index dd3a28e..f123ac7 100644
--- a/extensions/cpsection/network/view.py
+++ b/extensions/cpsection/network/view.py
@@ -883,6 +883,9 @@ class Network(SectionView):
             setting.apply()
         self._model.set_proxy_profile_name(self._proxy_profile_name)
 
+        if self._hidden_conn_manager.enabled:
+            self.__connect_hidden_net_cb(None)
+
     def _validate(self):
         if self._jabber_valid and self._radio_valid:
             self.props.is_valid = True
-- 
1.8.1.4

