From 85eab4998a28903409fc39ee51205eacfefec9a1 Mon Sep 17 00:00:00 2001
From: Gonzalo Odiard <godiard@gmail.com>
Date: Tue, 13 May 2014 14:16:48 -0300
Subject: [PATCH 1/2] hidden networks: Show password as asterisks - part of
 XOS-60

---
 extensions/cpsection/network/view.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/extensions/cpsection/network/view.py b/extensions/cpsection/network/view.py
index 1758493..e8ec381 100644
--- a/extensions/cpsection/network/view.py
+++ b/extensions/cpsection/network/view.py
@@ -471,6 +471,8 @@ class Network(SectionView):
                                     style.DEFAULT_SPACING)
                                 setting_box.set_spacing(style.DEFAULT_SPACING)
                                 entry = Gtk.Entry()
+                                if 'password' in key:
+                                    entry.set_visibility(False)
                                 setting_box.pack_start(entry, True, True, 0)
                                 box.pack_start(setting_box, True, True, 0)
                                 self._entries_properties[entry] = key
-- 
1.9.0

