From 3b6187eada640b7acbe51c38ae8fe6aaf4f65c48 Mon Sep 17 00:00:00 2001
From: Gonzalo Odiard <godiard@gmail.com>
Date: Thu, 15 May 2014 15:27:54 -0300
Subject: [PATCH 2/3] Fix KeyDialog password visibility - XOS-60

Signed-off-by: Gonzalo Odiard <godiard@sugarlabs.org>
---
 src/jarabe/desktop/keydialog.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/jarabe/desktop/keydialog.py b/src/jarabe/desktop/keydialog.py
index 54450fa..d5286fe 100644
--- a/src/jarabe/desktop/keydialog.py
+++ b/src/jarabe/desktop/keydialog.py
@@ -320,6 +320,7 @@ class KeyDialog(Gtk.Dialog):
 
     def add_key_entry(self):
         self._entry = Gtk.Entry()
+        self._entry.set_visibility(False)
         self._entry.connect('changed', self._update_response_sensitivity)
         self._entry.connect('activate', self._entry_activate_cb)
         self.vbox.pack_start(self._entry, True, True, 0)
-- 
1.9.0

