From 26366154db56c61ae17d38859064a3701dfe6eda Mon Sep 17 00:00:00 2001
From: Gonzalo Odiard <godiard@gmail.com>
Date: Fri, 20 Sep 2013 13:04:29 -0300
Subject: [PATCH 2/4] Make combo in proxy selection more generic

To enable the use on the hidden network configuration.

Signed-off-by: Gonzalo Odiard <gonzalo@laptop.org>
---
 extensions/cpsection/network/view.py | 50 ++++++++++++++++++++++++------------
 1 file changed, 34 insertions(+), 16 deletions(-)

diff --git a/extensions/cpsection/network/view.py b/extensions/cpsection/network/view.py
index 51f22ff..73ecd0f 100644
--- a/extensions/cpsection/network/view.py
+++ b/extensions/cpsection/network/view.py
@@ -58,7 +58,7 @@ class SettingBox(Gtk.HBox):
 class ComboSettingBox(Gtk.VBox):
 
     __gsignals__ = {
-        'profile-selected': (GObject.SignalFlags.RUN_FIRST, None, ([object])),
+        'changed': (GObject.SignalFlags.RUN_FIRST, None, ([])),
     }
 
     """
@@ -77,17 +77,17 @@ class ComboSettingBox(Gtk.VBox):
         setting_box.show()
 
         model = Gtk.ListStore(str, str, object, object)
-        self._combo_box = Gtk.ComboBox(model=model)
-        self._combo_box.connect('changed', self.__combo_changed_cb)
-        setting_box.pack_start(self._combo_box, True, True, 0)
-        self._combo_box.show()
+        self.combo_box = Gtk.ComboBox(model=model)
+        self.combo_box.connect('changed', self.__combo_changed_cb)
+        setting_box.pack_start(self.combo_box, True, True, 0)
+        self.combo_box.show()
 
         cell_renderer = Gtk.CellRendererText()
         cell_renderer.props.ellipsize = Pango.EllipsizeMode.MIDDLE
         cell_renderer.props.ellipsize_set = True
-        self._combo_box.pack_start(cell_renderer, True)
-        self._combo_box.add_attribute(cell_renderer, 'text', 0)
-        self._combo_box.props.id_column = 1
+        self.combo_box.pack_start(cell_renderer, True)
+        self.combo_box.add_attribute(cell_renderer, 'text', 0)
+        self.combo_box.props.id_column = 1
 
         self._settings_box = Gtk.VBox()
         self._settings_box.show()
@@ -96,12 +96,29 @@ class ComboSettingBox(Gtk.VBox):
         for optset in option_sets:
             model.append(optset)
 
-        setting.bind(setting_key, self._combo_box, 'active-id',
+        setting.bind(setting_key, self.combo_box, 'active-id',
                      Gio.SettingsBindFlags.DEFAULT)
 
     def __combo_changed_cb(self, combobox):
-        giter = combobox.get_active_iter()
-        new_box = combobox.get_model().get(giter, 2)[0]
+        self.emit('changed')
+
+
+class ProxyModeCombo(ComboSettingBox):
+
+    __gsignals__ = {
+        'profile-selected': (GObject.SignalFlags.RUN_FIRST, None, ([object])),
+    }
+
+    def __init__(self, name, setting, setting_key,
+                 option_sets, size_group=None):
+        ComboSettingBox.__init__(self, name, setting, setting_key,
+                 option_sets, size_group)
+
+        self.connect('changed', self.__combo_changed_cb)
+
+    def __combo_changed_cb(self, combo_setting_box):
+        giter = combo_setting_box.combo_box.get_active_iter()
+        new_box = combo_setting_box.combo_box.get_model().get(giter, 2)[0]
         current_box = self._settings_box.get_children()
         if current_box:
             self._settings_box.remove(current_box[0])
@@ -109,16 +126,17 @@ class ComboSettingBox(Gtk.VBox):
         self._settings_box.add(new_box)
         new_box.show()
 
-        profile = combobox.get_model().get(giter, 3)[0]
+        profile = combo_setting_box.combo_box.get_model().get(giter, 3)[0]
         self.emit('profile-selected', profile)
 
     def set_active(self, active):
-        self._combo_box.set_active(active)
-        giter = self._combo_box.get_active_iter()
-        profile = self._combo_box.get_model().get(giter, 3)[0]
+        self.combo_box.set_active(active)
+        giter = self.combo_box.get_active_iter()
+        profile = self.combo_box.get_model().get(giter, 3)[0]
         self.emit('profile-selected', profile)
 
 
+
 class OptionalSettingsBox(Gtk.VBox):
     """
     Container for settings (de)activated by a top-level setting.
@@ -501,7 +519,7 @@ class Network(SectionView):
             index = len(option_sets) - 1
             proxy_profiles_index[proxy_profile['title']] = index
 
-        box_mode = ComboSettingBox(
+        box_mode = ProxyModeCombo(
             _('Method:'), self._proxy_settings['org.gnome.system.proxy'],
             'mode', option_sets, size_group)
         box_mode.connect('profile-selected', self._apply_proxy_profile)
-- 
1.8.1.4

