From 9381aa84e6725fad0d58eb2a1a0fba8ba5f2cac8 Mon Sep 17 00:00:00 2001
From: Gonzalo Odiard <godiard@gmail.com>
Date: Wed, 28 Aug 2013 15:12:07 -0300
Subject: [PATCH] Inhibit suspend while sharing - OLPC #10363

Signed-off-by: Simon Schampijer <erikos@laptop.org>
---
 src/sugar/activity/activity.py | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/src/sugar/activity/activity.py b/src/sugar/activity/activity.py
index a0e2e11..f8cf2f9 100644
--- a/src/sugar/activity/activity.py
+++ b/src/sugar/activity/activity.py
@@ -97,6 +97,8 @@ J_DBUS_SERVICE = 'org.laptop.Journal'
 J_DBUS_PATH = '/org/laptop/Journal'
 J_DBUS_INTERFACE = 'org.laptop.Journal'
 
+POWERD_INHIBIT_DIR = '/var/run/powerd-inhibit-suspend'
+
 CONN_INTERFACE_ACTIVITY_PROPERTIES = 'org.laptop.Telepathy.ActivityProperties'
 
 
@@ -735,6 +737,18 @@ class Activity(Window, gtk.Container):
         else:
             self._jobject.metadata['share-scope'] = SCOPE_NEIGHBORHOOD
 
+    def _inhibit_suspend(self):
+        if not os.path.exists(POWERD_INHIBIT_DIR):
+           return
+
+        path = os.path.join(POWERD_INHIBIT_DIR, str(os.getpid()))
+        try:
+            fd = open(path, 'w')
+        except IOError:
+            logging.error("Inhibit Suspend: Could not create file %s", path)
+        else:
+            fd.close()
+
     def __joined_cb(self, activity, success, err):
         """Callback when join has finished"""
         logging.debug('Activity.__joined_cb %r', success)
@@ -744,6 +758,8 @@ class Activity(Window, gtk.Container):
             logging.debug('Failed to join activity: %s', err)
             return
 
+        self._inhibit_suspend()
+
         self.reveal()
         self.emit('joined')
         self.__privacy_changed_cb(self.shared_activity, None)
@@ -772,6 +788,8 @@ class Activity(Window, gtk.Container):
 
         activity.props.name = self._jobject.metadata['title']
 
+        self._inhibit_suspend()
+
         self.shared_activity = activity
         self.shared_activity.connect('notify::private',
                 self.__privacy_changed_cb)
@@ -886,6 +904,11 @@ class Activity(Window, gtk.Container):
         # Make the exported object inaccessible
         dbus.service.Object.remove_from_connection(self._bus)
 
+        if os.path.exists(POWERD_INHIBIT_DIR):
+            path = os.path.join(POWERD_INHIBIT_DIR, str(os.getpid()))
+            if os.path.exists(path):
+                os.unlink(path)
+
         self._session.unregister(self)
 
     def close(self, skip_save=False):
-- 
1.8.1.4

